avpipe.go

```diff
@@ -275,6 +276,30 @@ func NewXcParams() *XcParams {
 	}
 }
 
+// Custom unmarshalJSON for XcParams to make things backwards compatible with prior serialization
```

Comment:
I would say explicitly what we do: allow parsing of both an audio index array of the correct size, or a larger audio index array accompanied by an n_audio field (for backward compatibility)